crates/router/src/core/webhooks.rs

```diff
@@ -847,27 +849,77 @@ pub async fn trigger_webhook_to_merchant<W: types::OutgoingWebhookType>(
 }
 
 pub async fn webhooks_wrapper<W: types::OutgoingWebhookType, Ctx: PaymentMethodRetrieve>(
+    flow: &impl router_env::types::FlowMetric,
     state: AppState,
     req: &actix_web::HttpRequest,
     merchant_account: domain::MerchantAccount,
     key_store: domain::MerchantKeyStore,
     connector_name_or_mca_id: &str,
     body: actix_web::web::Bytes,
 ) -> RouterResponse<serde_json::Value> {
-    let (application_response, _webhooks_response_tracker) = webhooks_core::<W, Ctx>(
-        state,
+    let start_instant = Instant::now();
+    let (application_response, webhooks_response_tracker, serialized_req) =
+        webhooks_core::<W, Ctx>(
+            state.clone(),
+            req,
+            merchant_account.clone(),
+            key_store,
+            connector_name_or_mca_id,
+            body.clone(),
+        )
+        .await?;
+
+    let request_duration = Instant::now()
+        .saturating_duration_since(start_instant)
+        .as_millis();
+
+    let serialized_request = masking::masked_serialize(&serialized_req)
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Could not convert webhook effect to string")?;
+
+    let request_id = RequestId::extract(req)
+        .await
+        .into_report()
+        .attach_printable("Unable to extract request id from request")
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+    let auth_type = auth::AuthenticationType::WebhookAuth {
+        merchant_id: merchant_account.merchant_id.clone(),
+    };
+    let status_code = 200;
+    let api_event = ApiEventsType::Webhooks {
+        connector: connector_name_or_mca_id.to_string(),
+        payment_id: webhooks_response_tracker.get_payment_id(),
+    };
+    let response_value = serde_json::to_value(&webhooks_response_tracker)
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Could not convert webhook effect to string")?;
+
+    let api_event = ApiEvent::new(
+        flow,
+        &request_id,
+        request_duration,
+        status_code,
+        serialized_request,
+        Some(response_value),
+        auth_type,
+        api_event,
         req,
-        merchant_account,
-        key_store,
-        connector_name_or_mca_id,
-        body,
-    )
-    .await?;
+    );
+    match api_event.clone().try_into() {
+        Ok(event) => {
+            state.event_handler().log_event(event);
+        }
+        Err(err) => {
+            logger::error!(error=?err, event=?api_event, "Error Logging API Event");
+        }
+    }
 
     Ok(application_response)
 }
 
-#[instrument(skip_all)]
+// #[instrument(skip_all)]
```

Comment:
did you mean to comment this^?

Reply:
added this back,
had commented this since the instrument macro adds a `'static` requirement on all the return types, which does not work with `Box<dyn erased_serde::Serialize + 'static>` 
have changed the return type to `serve_json::Value` which would be the masked json value